velox/exec/Window.cpp

```diff
@@ -187,6 +195,28 @@ void Window::createWindowFunctions() {
   }
 }
 
+// The supportRankWindowBuild is designed to support 'rank' and
```

Comment:
This optimization can be applied more broadly. With default frame, all aggregate functions can be processed in streaming manner without holding all rows of a partition in memory. 

See code around analyzeFrameValues in AggregateWindow.cpp for some context.

Also, note that row_number and rank functions ignore frames.